### DISCUSSION TITLE
bug: X-GitHub-Api-Version not allowed with CORS

### DISCUSSION THREAD

**Discussion by baywet** [community/community#40619] [Apps, API and Webhooks] [answered]
### Select Topic Area

Bug

### Body

[A new header was recently introduced to enable API versioning](https://github.blog/2022-11-28-to-infinity-and-beyond-enabling-the-future-of-githubs-rest-api-with-api-versioning/) that's a great step forward.
The issue however is that header hasn't been allowed in the list of allowed headers for cross origin requests. **This effectively means the browser will block requests if the version is specified!!!**

In technical terms
```http
OPTIONS https://api.github.com/search/repositories?q=github%20topic%3Aopenapi-index&page=1'
authority:api.github.com
method:OPTIONS
path:/search/repositories?q=github%20topic%3Aopenapi-index&page=1
scheme:https
accept:*/*
accept-encoding:gzip, deflate, br
accept-language:en-US,en;q=0.9,fr;q=0.8
access-control-request-headers:x-github-api-version
access-control-request-method:GET
origin:https://app.kiota.dev
referer:https://app.kiota.dev/
sec-fetch-dest:empty
sec-fetch-mode:cors
sec-fetch-site:cross-site
```

Will give you

```
204 NoContent
Access-Control-Expose-Headers: ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset
```

(other headers ignored for brevity) 

when it should give you instead

```
Access-Control-Expose-Headers: ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset, X-GitHub-Api-Version 
```

**Comment by timrogers** [comment, marked as answer]
Hi @baywet ! 👋 Thanks for flagging this. We're already aware of the issue and are working on it. We'll update here once we've updated the CORS headers.

**Reply by baywet** [reply]
thanks for the follow up. Any ETA for the fix?

**Reply by CallumWatkins** [reply]
@timrogers 4 months later, can we get an update?

**Comment by mikejar** [comment]
Any update on this?

**Comment by statusunknown418** [comment]
has someone fixed this yet (or found any workaround?)

**Reply by baywet** [reply]
It's not fixed yet as far as I can tell and there's no workaround since the browser won't allow the request after the preflight
